app/routes/__auth/user.server.ts

```diff
@@ -14,8 +16,59 @@ export async function getUser({ headers }: Request) {
   const email = session.get('email') as string
   const groups = session.get('groups') as string[]
   const idp = session.get('idp') as string | null
-
+  const refreshToken = session.get('refreshToken') as string
+  const cognitoUserName = session.get('cognitoUserName') as string
+  const token = session.get('token') as string
   if (!sub) return null
+  const user = { sub, email, groups, idp, refreshToken, cognitoUserName, token }
+  if (
+    !token ||
+    JSON.parse(Buffer.from(token.split('.')[1], 'base64').toString()).exp *
```

Comment:
I think that we already have a dependency on jose. Use it to parse the JWT.

Reply:
Is there a specific method that handles this with jose? I've never heard of that package

Reply:
jose parses JWTs. It is a dependency of node-openid-client.

Reply:
Is this what you're talking about using? https://github.com/panva/jose/blob/HEAD/docs/functions/jwt_verify.jwtVerify.md#readme
If so, where can I find the secret, issuer, and audience values?

Reply:
Or are we able to just do this through one of the issue or client classes from node-openid-client?

Reply:
Possibly disregard, I updated it by parsing the JSON and using the TokenSet class's expired method